### PR TITLE
Search Tickets from CLI

### PR DESCRIPTION
**Pull request by mburggraf**
Added a small CLI tool to execute a basic ticket search from the command line (state, title, queue, number).

### CODE REVIEW

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+    $Self->Description('Search Tickets.');
+    $Self->AddOption(
+        Name        => 'count',
+        Description => "Count matching tickets.",
+        Required    => 0,
+        HasValue    => 0,
+    );
+    $Self->AddOption(
+        Name        => 'userid',
```
Comment (mgruner) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: For the options, I'd prefer a - as separator, such as user-id, queue-ids and so on.

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    );
+    $Self->AddOption(
+        Name        => 'title',
+        Description => "Ticket title.",
+        Required    => 0,
+        HasValue    => 1,
+        ValueRegex  => qr/^.*$/smx,
+    );
+    $Self->AddOption(
+        Name        => 'ticketnum',
```
Comment (mgruner) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: ticket-number please

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        Required    => 0,
+        HasValue    => 1,
+        ValueRegex  => qr/^(\d|\*|%)+(,(\d|\*|%)+)*$/smx,
+    );
+    $Self->AddOption(
+        Name        => 'queues',
+        Description => "Queus to search in.",
+        Required    => 0,
+        HasValue    => 1,
+        ValueRegex  => qr/^.*$/smx,
```
Comment (mgruner) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: In this and other cases with multiple values, I'd suggest to rather use the `Multiple => 1` parameter and call the option just queue, queue-id and so on. `--queue-id 1 --queue-id 2` etc. That avoids the need to split the value.

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+    if ( $Self->{ResultType} eq 'COUNT' ) {
+        $Self->Print(" Number of matching tickets: ".$result[0]."\n");
+    }
+    else {
+        if ( !@result ) {
+            $Self->Print("<red>No results.</red>\n");
+        }
+        else {
+            $Self->Print(" ".join("\n ", @result)."\n");
```
Comment (mgruner) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: Perhaps we should also add an option --show-details which will print one ticket per line, with TicketID, TicketNumber and Title?

**Comment by mgruner** [comment]
Hello @mburggraf, can you let me know if you plan to update this PR please?

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm (hunk 3 above)
Comment (mburggraf) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: Currently I don't have much time.
This definately sounds better than the current way (would probably have used it right away if the Multiple-Option was documented)

File: Kernel/System/Console/Command/Maint/Ticket/Search.pm (hunk 4 above)
Comment (mburggraf) [inline Kernel/System/Console/Command/Maint/Ticket/Search.pm]: My original thought was to keep it slim. To get more details by using the other shell tools.

**Comment by mgruner** [comment]
Hello @mburggraf, I'm closing this PR for now. Please feel free to apply the updates and then open another one for review!
Best regards, Martin

**Pull request closed without merging**